include/lm-observation-and-visualization.h

```diff
@@ -21,11 +22,11 @@ constexpr float LM_COLOR_BLUE = 255;
 /**
  * @brief Publishes visualization messages for landmarks in the global frame
  * @details Intended to be used with sampleLandmarks() in FastSLAMPF
- * @param[in] landmarks: queue of landmarks to be visualized
+ * @param[in] landmarks: vector of landmarks to be visualized
  * @param[in] landmark_visualization_publisher: publisher to publish the visualization messages
  */
 
-void visualizePFLandmarks(std::queue<Point2D> landmarks, 
+void visualizePFLandmarks(std::vector<Point2D> landmarks, 
```

Comment:
Pretty expensive to pass a vector in by value here. Please pass this by const reference.